src_cpp/opennnBridge/openNNnif.h

```diff
@@ -225,6 +226,23 @@ static ERL_NIF_TERM decode_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv
 
     //char* buf = (char*) enif_alloc(NumOfBytes + 1);
     
+    // TODO Ziv
```

Comment:
@zivmo99 
Ziv please read my notes within the code. 
The implementation is way more complex than needed and vulnerable. 
Please use the simple get and std::string as the example I gave.
I assume that erlang nif doesn't support templates so there are two options: 
1. Implement another template function and call it from within decode nif. 
2. Implement a switch case (better approach since you need to parse the atom to deduce the type).

Reply:
Messages optimization work will be done in another branch and PR 
https://github.com/leondavi/NErlNet/pull/167